### PR TITLE
iev edits

### PR DESCRIPTION
**Pull request by ivoysey**
Here are some thoughts about the things we talked about this afternoon!

### CODE REVIEW

**Comment by ivoysey** [comment]
also sorry for whitespace munging

File: abstract.tex [hunk 1]
```diff
@@ -14,7 +14,9 @@
 %
 association lists allow arbitrary order and may contain duplicates (unless refined with an explicit proof about validity), and
 %
-partial functions cannot be destructed and their equality is not decidable.
+partial functions cannot be destructed and their equality is not decidable
+in most theories without postulating function extensionality, which may or
```
Comment (nickcollins) [inline abstract.tex:18]: Equality of functions is never decidable, even with funext, right? deciding function equality isn't computable, right? With funext, we can state that two functions are equal if we have a proof that they behave equivalently, but we have no way of deciding that two arbitrary functions are unequal, thus we lack decidabliity

File: intro.tex [hunk 2]
```diff
@@ -8,9 +8,26 @@ \section{Introduction}
 %
 As such, proof assistants often use entirely different data structures than those of conventional settings, with a focus on simplicity or useful proof-theoretic properties.
 
-Consider dictionaries, which---among a broad range of purposes---are often used for implementing type contexts and evaluation environments.
-%
-The most basic representation for a dictionary, an \emph{\sal}, comprises a list of key-value pairs~\citep[Lists]{Pierce:SF1}.
+Consider dictionaries, which---among a broad range of purposes---are often
+used for implementing type contexts and evaluation environments. The
+contexts resulting from the selection of data structure to implement
```
Comment (nickcollins) [inline intro.tex:13]: Sorry, I've changed my mind - after rereading the intro, I think we should only lightly touch on the rationales, so as to keep the pacing tight, and to refer to section 3 for any detailed discussion of rationale. So I think we should briefly mention contraction and exchange at the end of the next paragraph (something like "This lack of \emph{\Extensional} can make proofs more difficult~\cite[Maps]{Pierce:SF1} or impossible, esp. when it comes to proving \emph{contraction} and \emph{exchange}. (TODO citation needed for contraction and exchange)"), and then you can merge the paragraph you have here (esp the second half, focusing on the general importance of contraction and exchange over the more specific question of how to implement dictionaries, which we've already discussed earlier) into the contraction and exchange paragraph in the Generality subsection of section 3

**Comment by nickcollins** [comment]
I'm going to go ahead and incorporate your suggestions into a separate commit and push it - may not be the ideal way to do things, but I'll need to make some tweaks on top of this stuff in order to maintain Ravi's formatting of the intro section.

**Pull request closed without merging**